### PR TITLE
mtt: add use of log; graphs: generation with seeds, min and max degree

### PR DESCRIPTION
**Pull request by chengguizi**
Changed two files:

mtt.py
- clean up numpy matrix usage
- add in log in calculating determinant, preventing overflow

graphs.py
- add option for feeding in seed, min and max degree during random graph gen

### CODE REVIEW

File: mtt.py [hunk 1]
```diff
@@ -45,23 +41,22 @@ def get_laplacian(adj_matrix):
         row[i] = degree
     return L
 
-                
-def MTT(graph):
+# return in natural log scale by default
```
Comment (pangjiahao) [inline mtt.py]: Did you mean default to be without log scale?

File: mtt.py [hunk 2]
```diff
@@ -116,3 +111,6 @@ def test():
     g5[11] = [7, 10, 12]
     g5[12] = [7, 11]
     print(MTT(g5) == 900)
+
+if __name__ == "__main__": 
+    test_mtt()
```
Comment (pangjiahao) [inline mtt.py:116]: The code looks wayyy better now!

**Review by pangjiahao — APPROVED** [review]
LGTM!

File: mtt.py (hunk 1 above)
Comment (chengguizi) [reply mtt.py]: Thanks, fixed the comment!

**Pull request merged**